### PR TITLE
Allow updating background color and font color on custom pages

### PR DESCRIPTION
**Pull request by taitus**
## References
Requirement:
> e. We would like to have custom colors for page headers

## Objectives
Allow define background color and font color for custom header on custom pages. 
In addition to `background_color` and `font_color` fields, we have also added a new field `custom_header_enabled` to allow set these fields (page with custom header) or create a custom page without custom header.

## Visual Changes
- Admin section:
With custom header enabled field unchecked does not render background color and font color fields

With custom header enabled field checked does render background color and font color fields

- Public page with custom header:
Public page with custom header

## Notes
We have to run this task after uploading to production:
`rake site_customization_pages:update_color_fields_values_from_deploy_yml`

This will update the background color and text color for the custom pages. The values for this task have been copied from the definition of the `deploy.yml` file.

### CODE REVIEW

File: app/views/custom/admin/site_customization/pages/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+  <div class="row">
+    <div class="column">
+      <div class="small-12 medium-6 large-3 column">
+        <%= f.check_box :custom_header_enabled %>
+      </div>
+    </div>
+  </div>
+
+  <div id="custom_header_colors" class="row <%= "hide" unless @page.custom_header_enabled? %>">
```
Comment (javierm) [inline app/views/custom/admin/site_customization/pages/_form.html.erb]:
```suggestion
  <div id="custom_header_colors" class="row">
```

Then, in the JavaScript, we add a call to `trigger()`:

```js
(function() {
  "use strict";
  App.SiteCustomizationPages = {
    initialize: function() {
      $("#site_customization_page_custom_header_enabled").on({
        change: function() {
          if (this.checked) {
            $("#custom_header_colors").removeClass("hide");
          } else {
            $("#custom_header_colors").addClass("hide");
          }
        }
      }).trigger("change");
    }
  };
}).call(this);
```

This way we slightly simplify the ERB code and there's no duplication between the ERB file and the JavaScript code (right now both are implementing code that says there should be a "hide" class when the checkbox isn't checked).

File: config/locales/custom/activerecord.yml [hunk 2]
```diff
@@ -26,4 +30,8 @@ en:
 
     attributes:
       budget/investment:
-        heading_id: "Category"
\ No newline at end of file
+        heading_id: "Category"
+      site_customization/page:
+        background_color: Background color
+        custom_header_enabled: "Enable custom header?"
```
Comment (javierm) [inline config/locales/custom/activerecord.yml]:
```suggestion
        custom_header_enabled: "Enable custom header"
```

I think we usually don't add question marks to texts for boolean fields :thinking:.

File: config/locales/custom/admin.yml [hunk 3]
```diff
@@ -196,3 +196,10 @@ ro:
         create: Creează o nouă înregistrare de date locale
       new:
         creating: Creează o nouă înregistrare de date locale
+    shared:
+      color_help: Format hexadecimal
+
+en:
+  admin:
+    shared:
+      color_help: Hexadecimal format
```
Comment (javierm) [inline config/locales/custom/admin.yml]:
```suggestion
```

These texts are already in CONSUL, so maybe we can remove these lines :thinking:.

File: db/migrate/20220314144443_add_color_fields_to_site_customization_pages.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class AddColorFieldsToSiteCustomizationPages < ActiveRecord::Migration[5.2]
+  def change
+    add_column :site_customization_pages, :background_color, :text
+    add_column :site_customization_pages, :font_color, :text
```
Comment (javierm) [inline db/migrate/20220314144443_add_color_fields_to_site_customization_pages.rb]: I know banners are `text` as well :relieved:, but was wondering whether it makes sense :thinking:. A `string` field would probably be more appropriate considering the text is going to be just 7 characters long. What do you think? :thinking:

File: lib/tasks/site_customization_pages.rake [hunk 5]
```diff
@@ -0,0 +1,15 @@
+namespace :site_customization_pages do
+  desc "Recover background_color and font_color defined in deploy.yml"
+  task update_color_fields_values_from_deploy_yml: :environment do
+    SiteCustomization::Page.all.each do |page|
+      background_color = Rails.configuration.deploy.dig("custom_pages", page.slug, "bgcolor")
+      font_color = Rails.configuration.deploy.dig("custom_pages", page.slug, "color") || "#FFF"
```
Comment (javierm) [inline lib/tasks/site_customization_pages.rake]: Why `|| "#FFF"`? Is there a case where the background color is defined but the font color isn't defined? :thinking:

File: spec/system/custom/admin/site_customization/pages_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+require "rails_helper"
+
+describe "Admin custom pages", :admin do
+  context "Create" do
+    scenario "Allow create custom page with color fields" do
```
Comment (javierm) [inline spec/system/custom/admin/site_customization/pages_spec.rb:5]: How about a scenario for editing existing pages? That way we can test the color fields are shown when editing a page with an enabled custom header.

File: spec/system/custom/site_customization/custom_pages_spec.rb [hunk 7]
```diff
@@ -0,0 +1,33 @@
+require "rails_helper"
+
+describe "Custom Pages" do
+  context "New custom page" do
```
Comment (javierm) [inline spec/system/custom/site_customization/custom_pages_spec.rb]: Not really sure what this line is about :thinking:; maybe we can remove it?

File: spec/system/custom/site_customization/custom_pages_spec.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    context "Custom header" do
```
Comment (javierm) [inline spec/system/custom/site_customization/custom_pages_spec.rb]:
```suggestion
    describe "Custom header" do
```

**Review by javierm — COMMENTED** [review]
@taitus I've left some comments. Let me know what you think!

**Review by javierm — APPROVED** [review]

**Comment by javierm** [comment]
@decembrya Could you review the [texts we've added in Romanian](https://github.com/code4romania/consul/pull/45/files#diff-693e522a555db173e5cf81b448791f7cbe305586aa220984eff2c9fcc9333da4R19)? Thanks! :pray:

**Comment by decembrya** [comment]
Translations look good 👍

**Pull request merged**